azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/ConfigMojo.java

```diff
@@ -0,0 +1,342 @@
+    }
+
+    private void configurePublicParent() throws IOException {
+        publicProjects = prompt.promoteMultipleEntities("project", targetProjects, t -> t.getName(), true, " to select none",
+                new ArrayList<MavenProject>());
+    }
+
+    private void configureAppName() throws IOException {
+        appName = prompt.promoteString("app name", "${project.artifactId}", "[A-Za-z0-9_\\.]+", true);
+    }
+
+    private void selectAppCluster() throws IOException, NoResourcesAvailableException {
+        final List<AppClusterResourceInner> clusters = SpringServiceUtils.getAvailableClusters(this.subscriptionId);
+        final AppClusterResourceInner clusterByName = clusters.stream().filter(t -> StringUtils.equals(this.clusterName, t.name())).findFirst()
+                .orElse(null);
+        if (clusterByName == null) {
+            clusterName = prompt.promoteSingleEntity("service", clusters, Utils.firstOrNull(clusters), t -> t.name(), true).name();
+        }
+
+        getLog().info(String.format("Using service: %s", TextUtils.blue(clusterName)));
+    }
+
+    private void initializeCredentials() throws InvalidConfigurationException, IOException, NoResourcesAvailableException {
+        // TODO: getAzureTokenCredentials will check auth for null, but maven will always map a default AuthConfiguration
+        final AzureTokenCredentials cred = AzureAuthHelper.getAzureTokenCredentials(this.auth);
+        azure = Azure.configure().authenticate(cred);
+        if (StringUtils.isBlank(subscriptionId)) {
+            subscriptionId = StringUtils.isBlank(cred.defaultSubscriptionId()) ? selectSubscription() : cred.defaultSubscriptionId();
+        }
+    }
+
+    private String selectSubscription() throws IOException, NoResourcesAvailableException {
+        final PagedList<Subscription> subscriptions = azure.subscriptions().list();
+        final Subscription select = this.prompt.promoteSingleEntity("subscription", subscriptions, Utils.firstOrNull(subscriptions),
+            t -> String.format("%s (%s)", t.displayName(), t.subscriptionId()), true);
+
+        return select.subscriptionId();
+    }
+
+    private boolean isProjectConfigured(MavenProject proj) {
+        final String pluginIdentifer = plugin.getPluginLookupKey();
+        final Xpp3Dom configuration = MavenUtils.getPluginConfiguration(proj, pluginIdentifer);
+        final List<String> topLevelProperties = Arrays.asList("subscriptionId", "appName", "isPublic", "runtimeVersion");
+
+        if (configuration == null) {
+            return false;
+        }
+
+        for (final Xpp3Dom child : configuration.getChildren()) {
+            if (topLevelProperties.contains(child.getName())) {
+                return true;
+            }
+        }
+
+        final List<String> deploymentProperties = Arrays.asList("vCPU", "memoryInGB", "instanceCount", "jvmOptions");
+        if (configuration.getChild(DEPLOYMENT_TAG) != null) {
```

Comment:
Shall we just check whether deployment tag is exist?

Reply:
Check deployment is not safe, since user may want to comment some existing configurations and see what is generated by config goal.